satellite/metainfo/batch.go

```diff
@@ -25,6 +25,9 @@ func (endpoint *Endpoint) Batch(ctx context.Context, req *pb.BatchRequest) (resp
 		switch singleRequest := request.Request.(type) {
```

Comment:
Would the batching logic be still valid when there are multiple headers. For example we had the idea of carrying forward some of the response values into next requests. Although, it seems we haven't implemented those. Supporting per batch item header, could make it more difficult to implement. Thoughts, @mniewrzal 

Maybe an extra check to ensure that the headers match the batch request?

Reply:
> Can you elaborate more about why it will be more difficult? Maybe I'm missing something.

Let's say I have a two different batch items, with different api keys. One creates an object and the other creates a segment. The second one doesn't have the stream id, so it needs to be gotten from the response for the first item... Now, technically we could say that all batch items act as invoked individually, with the exception of some fields that are rolled forward from previous responses when they are zero.

So, yeah, probably not more complicated, wondering whether it introduces some potential issues down the road.

Reply:
I see. I general, until it would be a valid case, I think it would be clearer and easier to support only batch items with the same api keys.

Reply:
I'll change the behavior to unconditionally set the individual operation header to the batch header and update the proto comment

Reply:
Then we can also remove probably the setting of the items when creating the batch items.

Reply:
oh true

Reply:
oh wait we already do though

Reply:
toRequest(nil) already gets passed nil for the header